src/input/findEditorConfiguration.ts

```diff
@@ -0,0 +1,35 @@
+import { FileSystem } from "../adapters/fileSystem";
+import { EditorConfiguration } from "./editorConfiguration";
+import { DeepPartial } from "./findReportedConfiguration";
+import { DEFAULT_VSCODE_SETTINGS_PATH } from "./vsCodeSettings";
+
+export type FindEditorConfigurationDependencies = {
+    fileSystem: Pick<FileSystem, "readFile">;
+};
+
+export const findEditorConfiguration = async (
+    dependencies: FindEditorConfigurationDependencies,
+    config: string | undefined,
+): Promise<DeepPartial<EditorConfiguration> | Error> => {
+    const rawConfiguration = await readConfiguration<EditorConfiguration>(
+        dependencies.fileSystem,
+        config || DEFAULT_VSCODE_SETTINGS_PATH,
+    );
+
+    return rawConfiguration;
+};
+
+const readConfiguration = async <Configuration>(
+    fileSystem: Pick<FileSystem, "readFile">,
+    path: string,
+): Promise<DeepPartial<Configuration> | Error> => {
+    try {
+        const fileContents = await fileSystem.readFile(path);
+        if (fileContents instanceof Error) {
+            throw new Error(fileContents.message);
+        }
+        return JSON.parse(fileContents) as DeepPartial<Configuration>;
```

Comment:
Ah, but many editor settings include comments -- even including [TSLint's](https://github.com/palantir/tslint/blob/master/.vscode/settings.json). This PR crashes as is for technically invalid JSON such as `//` comments. 

`findRawConfiguration` gets around it by using `import` (via `importer`). That should work for _importing_ the contents... but then you'd get the issue of comments being removed. That seems "fine" as a followup issue; adding (or finding) some kind of fancy diff-based output (or using TypeScript's?!) seems... hard.

Reply:
Changed it to use `findRawConfiguration`. Works well as described and is less custom code. Thanks.